pkgs/applications/misc/st/xst.nix

```diff
@@ -0,0 +1,34 @@
+{ stdenv, fetchurl, pkgconfig, libX11, ncurses, libXext, libXft , fontconfig
+, patches ? [], extraLibs ? []}:
+
+with stdenv.lib;
+
+let
+  version = "0.7.1";
+  name = "xst-${version}";
+  patches' = if patches == null then [] else patches;
```

Comment:
The question is, is `patches` really necessary?
You can always use:

```
xst.overrideDerivation (old: {
  patches = [ ./fancy.patch ];
})
```

in packagesOverrides or directly in `systemPackages`. 

Same for `extraLibs`:

```
xst.overrideDerivation (old: {
  buildInputs = [ libfancy ] ++ old.buildInputs;
})
```